sroka/api/google_ad_manager/gam_api.py

```diff
@@ -155,3 +155,65 @@ def get_users_from_admanager(query, dimensions, network_code=None):
     except errors.AdManagerReportError as e:
         print('Failed to generate user list. Error was: {}'.format(e))
         return
+
+def get_companies_from_admanager(query, dimensions, network_code=None):
+
+    company_df = pd.DataFrame()
+
+    dimensions_df = pd.DataFrame()
+
+    # .Where() handles filtering with admanager method, while statement_query handles queries with WHERE clause
+
+    statement_query = query.upper().replace("WHERE ", "")
```

Comment:
another thing to consider in the refactor would be type checker if the user provided correct types of arguments. See here for an example: https://github.com/Wikia/sroka/blob/master/sroka/api/neo4j/neo4j_api.py#L10
I'm mentioning it here because if I put int as a query here upper() function raises an error.

Reply:
Test cases:

```
query = []
dimensions = 123

data = get_companies_from_admanager(query, dimensions, network_code=5441)
data
# both variables should be called out
```

```
query = 123
dimensions = []

data = get_users_from_admanager(query, dimensions)
data
# query variable should be called out 
```

```
start_day = '01'
end_day = '30'
start_month = '07'
end_month = '09'
year = '2021'

query = ""
dimensions = {'LINE_ITEM_NAME':123}
custom_field_id = 123
columns = ['AD_SERVER_IMPRESSIONS']
start_date = ['year', year,
              'month', start_month,
              'day', start_day]
stop_date = {'year': year,
             'month': end_month,
             'day': end_day}

data = get_data_from_admanager(query, dimensions, columns, start_date, stop_date, custom_field_id)
# dimensions, custom_field_id and start_date should be called out
```

```
start_day = '01'
end_day = '30'
start_month = '07'
end_month = '09'
year = '2021'

query = 123
dimensions = {'LINE_ITEM_NAME':123}
custom_field_id = 123
columns = ['AD_SERVER_IMPRESSIONS']
start_date = ['year', year,
              'month', start_month,
              'day', start_day]
stop_date = {'year': year,
             'month': end_month,
             'day': end_day}

data = get_data_from_admanager(query, dimensions, columns, start_date, stop_date, custom_field_id, network_code="abc")
# query, dimensions, custom_field_id, start_date, network_code should be called out
```